produce/lib/produce/merchant.rb

```diff
@@ -5,60 +5,46 @@ class Merchant
     def create(_options, _args)
       login
 
-      merchant_identifier = Produce.config[:merchant_identifier] || UI.input("Merchant identifier (reverse-domain name style string starting with 'merchant'): ")
+      merchant_identifier = detect_merchant_identifier
 
-      if merchant_exists? merchant_identifier
-        UI.success("[DevCenter] Merchant '#{merchant_identifier})' already exists, nothing to do on the Dev Center")
+      if merchant = find_merchant(merchant_identifier)
+        UI.success("[DevCenter] Merchant '#{merchant.bundle_id})' already exists, nothing to do on the Dev Center")
       else
         merchant_name = Produce.config[:merchant_name] || merchant_identifier.split(".").map(&:capitalize).reverse.join(' ')
-
         UI.success("Creating new merchant '#{merchant_name}' with identifier '#{merchant_identifier}' on the Apple Dev Center")
-
-        merchant = Spaceship.merchant.create!(bundle_id: merchant_identifier,
-                                            name: merchant_name,
-                                            mac: Produce.config[:platform].to_s == 'mac')
+        merchant = Spaceship.merchant.create!(bundle_id: merchant_identifier, name: merchant_name, mac: self.class.mac?)
 
         if merchant.name != merchant_name
-          UI.important("Your merchant name includes non-ASCII characters, which are not supported by the Apple Developer Portal.")
+          UI.important("Your merchant name might include non-ASCII characters, which are not supported by the Apple Developer Portal.")
           UI.important("To fix this a unique (internal) name '#{merchant.name}' has been created for you.")
         end
 
         UI.message("Created merchant #{merchant.merchant_id}")
-        UI.user_error!("Something went wrong when creating the new merchant - it's not listed in the merchants list") unless merchant_exists? merchant_identifier
         UI.success("Finished creating new merchant '#{merchant_name}' on the Dev Center")
       end
-
-      return true
     end
 
     def associate(_options, args)
       login
 
-      if !app_exists?
-        UI.message("[DevCenter] App '#{Produce.config[:app_identifier]}' does not exist, nothing to associate with the merchants")
-      else
-        app = Spaceship.app.find(app_identifier)
-        UI.user_error!("Something went wrong when fetching the app - it's not listed in the apps list") if app.nil?
-
+      if app = Spaceship.app.find(app_identifier)
```

Comment:
@ohayon I've split the merchant name out into a separate method and broken it apart as well as added a test into the new `merchant_spec.rb` in my latest commit.